src/article/ArticleAPI.js

```diff
@@ -437,4 +437,37 @@ export default class ArticleAPI {
       documentHelpers.insertAt(tx, [figure.id, 'panels'], pos, node.id)
     })
   }
+
+  _getSurfaceId (node, path, viewName) {
```

Comment:
You could add a comment explaining your feelings about this implementation, and what the basic idea of the approach is

Reply:
e.g.: 
```
HACK: determining proper surfaceId in a hard-coded way.
Using templates for different use cases to derive ....
```

Reply:
The more you think a method is, the more important is a comment.